browser/src/robocorp/browser/_browser_context.py

```diff
@@ -122,24 +122,30 @@ def _browser_config() -> _BrowserConfig:
     return _BrowserConfig()
 
 
-def _is_debugger_attached() -> bool:
-    pydevd = sys.modules.get("pydevd")
-    if not pydevd or not hasattr(pydevd, "get_global_debugger"):
-        return False
-    debugger = pydevd.get_global_debugger()
-    if not debugger or not hasattr(debugger, "is_attached"):
-        return False
-    return debugger.is_attached()
+def _get_auto_headless_state() -> bool:
+    # If in Linux and with no valid display, we can assume we are in a
+    # container which doesn't support UI.
+    return sys.platform.startswith("linux") and not (
+        os.environ.get("DISPLAY") or os.environ.get("WAYLAND_DISPLAY")
+    )
 
 
 @session_cache
 def browser_type_launch_args() -> Dict:
     launch_options = {}
-    headless = _browser_config().headless
-    if headless is None:
-        launch_options["headless"] = not _is_debugger_attached()
+
+    # RPA_HEADLESS_MODE can be used to force whether running headless.
+    rpa_headless_mode = os.environ.get("RPA_HEADLESS_MODE")
```

Comment:
It _does_ look like workforce agent sets this in docker runs, which makes this behaviour good as the user can't break the bot by accident. Unless there is some use-case where the user explicitly tries to run headful using xvfb or similar.

I'm slightly annoyed by the fact that it's an envvar with `RPA_` prefix, since we've tried to avoid those in these new libraries in favor of Control Room's `RC_` prefix. I wonder if it would make sense to add a second one that workforce agent supports? `RC_BROWSER_HEADLESS`? Maybe not...

Reply:
I guess we should check with the platform guys -- I'm ok with adding it if it'll be used ;)

Reply:
Maybe there was a distinction between `RC_*` (what comes from Control Room / Robocorp) and `RPA_*` (what you set to control the automation, no matter where the bot is running). And sometimes the env var was overriding the explicit parameter, other times the parameter takes precedence.